ibis/backends/duckdb/__init__.py

```diff
@@ -341,11 +341,23 @@ def get_schema(
         """
         conditions = [sg.column("table_name").eq(sge.convert(table_name))]
 
-        if catalog is not None:
-            conditions.append(sg.column("table_catalog").eq(sge.convert(catalog)))
+        # If no catalog.database is specified, assume the current ones
+        db = database or self.current_database
+        cat = catalog or self.current_catalog
+
+        # If a catalog isn't specified, then include temp tables in the
+        # returned values
+        # temp tables live in the `temp` catalog
+        # We assume that we should return temp tables that match the database
+        # specified
+        if catalog is None:
+            conditions.append(
+                sg.column("table_catalog").isin(sge.convert(cat), sge.convert("temp"))
+            )
+        else:
+            conditions.append(sg.column("table_catalog").eq(sge.convert(cat)))
```

Comment:
Given this part of the PR description:

> When you ask for a table named foo, we will look through the current_database and wherever temp tables are stored for a match and return the schema of that table.

Shouldn't the implementation be this?

```suggestion
        conditions.append(
            sg.column("table_catalog").isin(sge.convert(cat), sge.convert("temp"))
        )
```

That is, always return temp tables?

Maybe I'm misreading and the fact that catalog is specified changes the desired output.

Reply:
This is going to have to be rewritten anyway, to only return tables from `temp` (or `main`) when there's a name collision

Reply:
Cleaned up this up to use `DESCRIBE catalog.database.table`, which will give the semantics of temp tables -> non temp tables.